File: taxi_sabana/debug_zones.dart

import 'dart:convert';
import 'dart:io';

String normalize(String s) {
   final replacements = {
     'á': 'a', 'é': 'e', 'í': 'i', 'ó': 'o', 'ú': 'u'
   };
   s = s.toLowerCase().trim();
   replacements.forEach((k, v) {
     s = s.replaceAll(k, v);
   });
   return s;
}

void resolve(String queryAddr, List<dynamic> barrios) {
  final query = normalize(queryAddr);
  print("");
  print("-" * 50);
  print("Address: $queryAddr");
  print("Normalized: $query");
  
  for (int i = 0; i < barrios.length; i++) {
    final b = barrios[i];
    final bNameRaw = (b['nombre_barrio'] ?? '').toString();
    final bName = normalize(bNameRaw);
    
    if (bName.isEmpty) continue;
    
    // Logic from ZonaResolver
    bool match = false;
    if (bName == query) match = true;
    else if (query.contains(bName)) match = true;
    else if (bName.contains(query)) match = true;
    
    if (match) {
      print("✅ MATCH FOUND at Index $i:");
      print("   Barrio: $bNameRaw");
      print("   Normalized Barrio: $bName");
      print("   Zone: ${b['zona']}");
      return; 
    }
  }
  print("❌ NO MATCH FOUND");
}

void main() async {
  final file = File('assets/sincelejo_barrios.json');
  final content = await file.readAsString();
  final List<dynamic> barrios = jsonDecode(content);
  
  final addresses = [
    "Cl. 43 A Bis # 17A-3 Majagual, Sincelejo, Sucre, Colombia",
    "Sincelejo, Colombia",
    "Majagual, Sincelejo",
    "Gran Colombia",
    "Cl 43 A Bis # 17A-3 Majagual"
  ];
  
  for (final a in addresses) {
    resolve(a, barrios);
  }
}

File: taxi_sabana/debug_zones.py
import json

def normalize(s):
    replacements = {
        'á': 'a', 'é': 'e', 'í': 'i', 'ó': 'o', 'ú': 'u'
    }
    s = s.lower().strip()
    for k, v in replacements.items():
        s = s.replace(k, v)
    return s

def resolve(query_addr):
    with open('e:/copia de seguridad/taxi_sabana/assets/sincelejo_barrios.json', 'r', encoding='utf-8') as f:
        barrios = json.load(f)
    
    query = normalize(query_addr)
    print(f"Address: {query_addr}")
    print(f"Normalized: {query}")
    
    for i, b in enumerate(barrios):
        bNameRaw = b.get('nombre_barrio', '')
        bName = normalize(bNameRaw)
        
        if not bName: continue
        
        # Exact logic from Dart:
        # if (bName == queryNorm || bName.contains(queryNorm) || queryNorm.contains(bName))
        
        match = False
        if bName == query: match = True
        elif bName in query: match = True # query.contains(bName)
        elif query in bName: match = True # bName.contains(query)
        
        if match:
            print(f"✅ MATCH FOUND at Index {i}:")
            print(f"   Barrio: {bNameRaw}") # Original name
            print(f"   Normalized Barrio: {bName}")
            print(f"   Zone: {b.get('zona')}")
            return

    print("❌ NO MATCH FOUND")

# Test cases
addresses = [
    "Cl. 43 A Bis # 17A-3 Majagual, Sincelejo, Sucre, Colombia",
    "Sincelejo, Colombia",
    "Majagual, Sincelejo",
    "Gran Colombia"
]

for a in addresses:
    print("-" * 50)
    resolve(a)
